### PR TITLE
Jack-in error messaging instead of silent failure when no shadow builds are selected

### LINKED ISSUE #2022: Add a clear message about selecting a build when jacking in with shadow-cljs fails because no build was selected
Sometimes users use the up and down arrow keys + `enter` to select a build, but this doesn't select a build because it's a checkbox menu. See [this comment](https://github.com/BetterThanTomorrow/calva/issues/2014?notification_referrer_id=NT_kwDOAMIiOLM1MjgyMTU0OTQ4OjEyNzIyNzQ0&notifications_query=repo%3ABetterThanTomorrow%2Fcalva#issuecomment-1383157315) and the issue description by @milahu.

We should bail out of jack-in gracefully and explain to the user that a build must be selected. I don't know if this just applies to shadow-cljs.

### PR DESCRIPTION
**Pull request by PEZ**
## What has changed?

* Catch and show error message in jack-in and copy jack-in command
   image

* Fixes #2022

## My Calva PR Checklist

I have:

- [x] Read [How to Contribute](https://github.com/BetterThanTomorrow/calva/wiki/How-to-Contribute#before-sending-pull-requests).
- [x] Directed this pull request at the `dev` branch. (Or have specific reasons to target some other branch.)
- [x] Made sure I have changed the PR base branch, so that it is not `published`. (Sorry for the nagging.)
- [x] Updated the `[Unreleased]` entry in `CHANGELOG.md`, linking the issue(s) that the PR is addressing.
- [x] Figured if **anything** about the fix warrants tests on Mac/Linux/Windows/Remote/Whatever, and either tested it there if so, or mentioned it in the PR.
- ~[ ] Added to or updated docs in this branch, if appropriate~
- [ ] Tests
  - [x] Tested the particular change
  - [x] Figured if the change might have some side effects and tested those as well.
  - [x] Smoke tested the extension as such.
  - [ ] Tested the VSIX built from the PR (so, after you've submitted the PR). You'll find the artifacts by clicking _Show all checks_ in the CI section of the PR page, and then _Details_ on the `ci/circleci: build` test.
- [x] Referenced the issue I am fixing/addressing _in a commit message for the pull request_.
  - [x] If I am fixing the issue, I have used [GitHub's fixes/closes syntax](https://help.github.com/en/articles/closing-issues-using-keywords)
- [ ] Formatted all JavaScript and TypeScript code that was changed. (use the [prettier extension](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) or run `npm run prettier-format`)
- [ ] Confirmed that there are no linter warnings or errors (use the [eslint extension](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint), run `npm run eslint` before creating your PR, or run `npm run eslint-watch` to eslint as you go).

Ping @pez, @bpringe, @corasaurus-hex, @Cyrik

### CODE REVIEW

File: src/nrepl/jack-in.ts [hunk 1]
```diff
@@ -270,9 +274,13 @@ export async function jackIn(connectSequence: ReplConnectSequence, cb?: () => un
...
-      const terminalJackInOptions = await getJackInTerminalOptions(projectConnectSequence);
-      if (terminalJackInOptions) {
-        executeJackInTask(terminalJackInOptions, projectConnectSequence, cb);
+      try {
+        const terminalJackInOptions = await getJackInTerminalOptions(projectConnectSequence);
+        if (terminalJackInOptions) {
+          executeJackInTask(terminalJackInOptions, projectConnectSequence, cb);
+        }
+      } catch (e) {
+        void vscode.window.showErrorMessage(`Error creating Jack-in command line: ${e}`, 'OK');
```
Comment (bpringe) [inline src/nrepl/jack-in.ts]:
```suggestion
        void vscode.window.showErrorMessage(`Error creating jack-in command line: ${e}`, 'OK');
```

File: src/nrepl/jack-in.ts [hunk 2]
```diff
@@ -144,10 +144,14 @@ export async function copyJackInCommandToClipboard(): Promise<void> {
...
   if (projectConnectSequence) {
-    const { executable, args } = await getJackInTerminalOptions(projectConnectSequence);
-    if (executable && args) {
-      void vscode.env.clipboard.writeText(createCommandLine(executable, args));
-      void vscode.window.showInformationMessage('Jack-in command copied to the clipboard.');
+    try {
+      const { executable, args } = await getJackInTerminalOptions(projectConnectSequence);
+      if (executable && args) {
+        void vscode.env.clipboard.writeText(createCommandLine(executable, args));
+        void vscode.window.showInformationMessage('Jack-in command copied to the clipboard.');
```
Comment (bpringe) [inline src/nrepl/jack-in.ts]:
```suggestion
        void vscode.window.showInformationMessage('Jack-in command line copied to the clipboard.');
```

**Review by bpringe — COMMENTED** [review]
Looks good to me! Just added a couple of suggestions. I like the wording "command line" versus "command". I think the command name and description should be updated to include "line" as well, but that doesn't have to be in this PR.

**Comment by PEZ** [comment]
> Looks good to me! Just added a couple of suggestions. I like the wording "command line" versus "command". I think the command name and description should be updated to include "line" as well, but that doesn't have to be in this PR.

We probably should change to command line consistently, if we decide to change this instance.

**Pull request merged**